nilearn/decomposition/multi_pca.py

```diff
@@ -164,11 +190,39 @@ def _raw_fit(self, data):
             S = np.sqrt(np.sum(data ** 2, axis=1))
             S[S == 0] = 1
             data /= S[:, np.newaxis]
-        self.components_, self.variance_, _ = self._cache(
-            randomized_svd, func_memory_level=2)(
-            data.T, n_components=self.n_components,
-            transpose=True,
-            random_state=self.random_state, n_iter=3)
+
+        self.components_, self.variance_ = self._group_pca(data)
+
         if self.do_cca:
             data *= S[:, np.newaxis]
-        self.components_ = self.components_.T
+
+    def _group_pca(self, data):
+        if self.incremental_group_pca:
+            components, variance = self._cache(
+                _incremental_group_pca, func_memory_level=2)(
+                    data, self.n_components, self._nb_subjects_)
+        else:
+            components, variance, _ = self._cache(
+                randomized_svd, func_memory_level=2)(
+                    data.T, n_components=self.n_components,
+                    transpose=True,
+                    random_state=self.random_state, n_iter=3)
+            components = components.T
+
+        return components, variance
+
+
+def _incremental_group_pca(subject_pcas, n_components, nb_subjects):
+    # delayed import because IncrementalPCA is only supported for
+    # sklearn versions > 0.16
+    from sklearn.decomposition.incremental_pca import IncrementalPCA
+
+    pca = IncrementalPCA(n_components=n_components)
+    subject_length = len(subject_pcas) // nb_subjects
+    subject_indices = (slice(i, i + subject_length)
```

Comment:
This is a bit hacky because now subject_pcas come in a single array instead of a list of per-subject PCA and the concactenation happens a bit deep in the BaseDecomposition code. I did that with the old code in mind but maybe I don't actually need to bother doing it ...